### PR TITLE
cri-o: honor additional and insecure registries again

### PR DESCRIPTION
**Pull request by giuseppe**
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by ashcrow** [comment]
bot, retest this please

File: roles/docker/tasks/systemcontainer_crio.yml [hunk 1]
```diff
@@ -1,18 +1,21 @@
 ---
 
 # TODO: Much of this file is shared with container engine tasks
+- set_fact:
```
Comment (michaelgugino) [inline roles/docker/tasks/systemcontainer_crio.yml]: Why are we setting these facts like this?

Let's not use set_fact for this, this is what role defaults are for.

File: roles/docker/tasks/systemcontainer_crio.yml [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+    l_insecure_crio_registries: ""
+    l_additional_crio_registries: ""
 - set_fact:
     l_insecure_crio_registries: "{{ '\"{}\"'.format('\", \"'.join(l2_docker_insecure_registries)) }}"
-  when: l2_docker_insecure_registries | bool
+  when: l2_docker_insecure_registries is defined
```
Comment (michaelgugino) [inline roles/docker/tasks/systemcontainer_crio.yml]: This variable is always defined, it's in role defaults.

**Comment by ashcrow** [comment]
> This role is being heavily refactored. These changes don't appear to be necessary.

@michaelgugino I believe @giuseppe is attempting to fix a regression that came from a refactoring regression previously.

**Comment by giuseppe** [comment]
@michaelgugino thanks for the review, I've done the simplifications you requested

**Comment by ashcrow** [comment]
Same unrelated failures noted in https://github.com/openshift/openshift-ansible/pull/6178

/cc @sdodson

**Comment by runcom** [comment]
@giuseppe maybe I'm asking for something not super easy, but can we have a unit/integration tests to make sure we don't regress anymore on this one?

/cc @sdodson @ashcrow

**Comment by giuseppe** [comment]
I don't think we can add unit test, but this was easy to catch if we had an installation configured with CRI-O, as AFAICS we use `additional_registries`.  This will be addressed once we will use CRI-O on each PR.

**Pull request merged**